embedchain/config/apps/BaseAppConfig.py

```diff
@@ -17,29 +28,43 @@ def __init__(self, log_level=None, embedding_fn=None, db=None, host=None, port=N
         :param id: Optional. ID of the app. Document metadata will have this id.
         :param host: Optional. Hostname for the database server.
         :param port: Optional. Port for the database server.
+        :param db_type: Optional. db type to use. Currently [chroma, es] are supported.
+        :param vector_dim: Vector dimension generated by embedding fn
         """
         self._setup_logging(log_level)
-
-        self.db = db if db else BaseAppConfig.default_db(embedding_fn=embedding_fn, host=host, port=port)
+        self.db = BaseAppConfig.get_db(
+            db=db, embedding_fn=embedding_fn, host=host, port=port, db_type=db_type, vector_dim=vector_dim
+        )
         self.id = id
         return
 
     @staticmethod
-    def default_db(embedding_fn, host, port):
+    def get_db(db, embedding_fn, host, port, db_type, vector_dim):
         """
-        Sets database to default (`ChromaDb`).
-
+        Get db based on db_type, db with default database (`ChromaDb`)
+        :param Optional. (Vector) database to use for embeddings.
         :param embedding_fn: Embedding function to use in database.
         :param host: Optional. Hostname for the database server.
         :param port: Optional. Port for the database server.
-        :returns: Default database
+        :param db_type: Optional. db type to use. Supported values (`es`, `chroma`)
+        :param vector_dim: Vector dimension generated by embedding fn
+        :returns: database instance
         :raises ValueError: BaseAppConfig knows no default embedding function.
         """
+        if db:
+            return VectorDb(db, db_type)
+
         if embedding_fn is None:
             raise ValueError("ChromaDb cannot be instantiated without an embedding function")
+
+        if db_type == "es":
+            from embedchain.vectordb.elasticsearch_db import EsDB
```

Comment:
also, i initialize my elasticsearch as 

```
ES_EXTRA_PARAMS = {
    "http_auth": (ES_USER, ES_PASSWORD),
    "ca_certs": ES_CA_CERTS,
}

ES_CONNECTION = Elasticsearch(ES_URL, **ES_EXTRA_PARAMS)
```

We should have a generic support so that a user can initialize ES in whatever way suits them.

Reply:
added `ElasticsearchDBConfig`  now we can initialize by 
`
es_config = ElasticsearchDBConfig(
	es_url='http://localhost:9200',
)
CustomApp(CustomAppConfig(
	embedding_fn=EmbeddingFunctions.OPENAI, 
	provider=Providers.OPENAI, 
	db_type=VectorDatabases.ELASTICSEARCH, 
	es_config=es_config,
))
`